### PR TITLE
[#AIO-2584] Validate max row limit based on file_origination_types

### PR DESCRIPTION
**Pull request by kcpavan**

### CODE REVIEW

**Review by tmking — APPROVED** [review]
👍Looks good!

File: spec/test_app/db/migrate/20220314124605_add_foreign_key_to_attachments_for_blob_id.active_storage.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # This migration comes from active_storage (originally 20180723000244)
-class AddForeignKeyConstraintToActiveStorageAttachmentsForBlobId < ActiveRecord::Migration[6.0]
+class AddForeignKeyToAttachmentsForBlobId < ActiveRecord::Migration[6.0]
```
Comment (kcpavan) [inline spec/test_app/db/migrate/20220314124605_add_foreign_key_to_attachments_for_blob_id.active_storage.rb:2]: I renamed this to fix (Gem::Package::TooLongFileName) error on `rake build` task.

File: spec/features/uploading_your_own_spreadsheet_spec.rb [hunk 2]
```diff
@@ -133,8 +133,14 @@
...
-    before { NfgCsvImporter.configuration.max_number_of_rows_allowed = 2 }
-    after { NfgCsvImporter.configuration.max_number_of_rows_allowed = 50000 }
+    before do
+      NfgCsvImporter.configuration.max_number_of_rows_allowed = 2
+      NfgCsvImporter.configuration.allowed_file_origination_types_to_bypass_max_row_limit = []
+    end
+    after do
+      NfgCsvImporter.configuration.max_number_of_rows_allowed = 20_000
+      NfgCsvImporter.configuration.allowed_file_origination_types_to_bypass_max_row_limit = []
+    end
```
Comment (joseluistorres) [inline spec/features/uploading_your_own_spreadsheet_spec.rb:144]:
```suggestion

    after do
      NfgCsvImporter.configuration.max_number_of_rows_allowed = 20_000
      NfgCsvImporter.configuration.allowed_file_origination_types_to_bypass_max_row_limit = []
    end
```

File: spec/forms/nfg_csv_importer/onboarding/import_data/upload_post_processing_form_spec.rb [hunk 3]
```diff
@@ -27,22 +27,32 @@
...
     it "should add errors to base" do
       subject
       expect(import_file_validateable_host.errors.messages[:base].first).to eq I18n.t('nfg_csv_importer.onboarding.import_data.invalid_number_of_rows', num_rows: 2)
     end
+
+    context 'when the file origination type is allowed to bypass the max row limit' do
+      before { NfgCsvImporter.configuration.allowed_file_origination_types_to_bypass_max_row_limit = [import.file_origination_type.type_sym] }
+      after { NfgCsvImporter.configuration.allowed_file_origination_types_to_bypass_max_row_limit = [] }
+
+      it "should not add errors to base" do
```
Comment (joseluistorres) [inline spec/forms/nfg_csv_importer/onboarding/import_data/upload_post_processing_form_spec.rb]:
```suggestion

      after { NfgCsvImporter.configuration.allowed_file_origination_types_to_bypass_max_row_limit = [] }

      it "does not add errors to base" do
```

File: spec/forms/nfg_csv_importer/onboarding/import_data/upload_post_processing_form_spec.rb [hunk 4]
```diff
@@ -27,22 +27,32 @@
...
+    end
   end
 
   context 'when number of rows is less than the maximum number of rows allowed' do
     before { NfgCsvImporter.configuration.max_number_of_rows_allowed = 10 }
-    after { NfgCsvImporter.configuration.max_number_of_rows_allowed = 50000 }
+    after { NfgCsvImporter.configuration.max_number_of_rows_allowed = 50_000 }
 
-    it "should add errors to base" do
+    it "should not add errors to base" do
```
Comment (joseluistorres) [inline spec/forms/nfg_csv_importer/onboarding/import_data/upload_post_processing_form_spec.rb]:
```suggestion
    it "does not add errors to base" do
```

**Review by joseluistorres — APPROVED** [review]
A couple of comments but no blockers!

**Comment by kcpavan** [comment]
Some import types bypass the upload_post_processing step, thereby skipping the existing row length validation. To address this, I have incorporated an additional validation during the preview_confirmation step to ensure comprehensive validation for all types(except NCOA and send_to_nfg import types).

**Review by tmking — APPROVED** [review]

**Pull request merged**